attractions/utils/call-on-sight.js

```diff
@@ -2,8 +2,9 @@
  * Action that allows calling a function when an element appears in the viewport.
  * Uses the IntersectionObserver API, which might not be available in older browsers like IE11.
  * If it is not available, the action will be silently ignored.
+ * @template T
  * @param {Element} node The HTML element to be observed
- * @param {{ callback: (...args: any[]) => void; args: any[] }} options Callback and its args
+ * @param {{ callback: (...args: T[]) => void; args: T[] }} options Callback and its args
```

Comment:
Doesn't that restrict people to arguments of just one type? Can't the template type be an array itself?

Reply:
It _kind of_ can in TS ([Variadic tuple type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types)), but not in [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#template).

Reply:
Also, they can always define `T` to be a custom interface and just use one parameter wrapped in a singleton array